sdb/commands/spl/internal/kmem_helpers.py

```diff
@@ -91,12 +92,23 @@ def object_size(cache: drgn.Object) -> int:
 def nr_objects(cache: drgn.Object) -> int:
     assert sdb.type_canonical_name(cache.type_) == 'struct spl_kmem_cache *'
     if backed_by_linux_cache(cache):
-        return int(cache.skc_obj_alloc.value_())
+        return obj_alloc(cache)
     return int(cache.skc_obj_total.value_())
 
 
 def obj_alloc(cache: drgn.Object) -> int:
     assert sdb.type_canonical_name(cache.type_) == 'struct spl_kmem_cache *'
+    try:
+        return int(drgn_percpu.percpu_counter_sum(cache.skc_linux_alloc))
+    except AttributeError:
+        #
+        # The percpu_counter referenced above wasn't in ZoL until the
+        # following commit: ec1fea4516ac2f0c08d31d6308929298d1b281d0
+        #
+        # Fall back to the old-mechanism if that percpu_counter member
+        # doesn't exist (an AttributeError will be thrown).
+        #
+        return int(cache.skc_obj_alloc.value_())
     return int(cache.skc_obj_alloc.value_())
```

Comment:
This line is equivalent to what we run in the `except AttributeError` case, so rather than duplicating the two, would it be better to use:
```
except AttributeError:
    pass
```
and let the code reach this line in that case?

Or perhaps we should more simply remove this line, since I think it's dead code?

Reply:
Actually thanks for pointing this out... this pointed out a bug in my code. We don't want to look at the percpu counter if this is not a linux_slub backed cache. I changed the code and retesting now.